bin/server.js

```diff
@@ -73,22 +78,32 @@ function prHook( event ) {
 						error: error.stack
 					});
 				});
+			failedEvents.push( event );
 		}
 	);
 }
 
 // Fetch new CLA signatures periodically
 getSignatures = (function() {
+	var signatures;
 	var promise = getHashedSignatures();
+	promise
+		.then(function( initialSignatures ) {
+			signatures = initialSignatures;
+		});
 
 	function updateSignatures() {
 		var updatedPromise = getHashedSignatures();
 
 		debug( "updating signatures" );
 		updatedPromise
-			.then(function() {
+			.then(function( newSignatures ) {
 				debug( "successfully updated signatures" );
 				promise = updatedPromise;
+				if ( Object.keys( signatures ).length !== Object.keys( newSignatures ).length ) {
+					signatures = newSignatures;
+					failedEvents.splice( 0 ).forEach( prHook );
```

Comment:
The list of failed events could potentially grow to a fairly large size over time. You might want to run these in series rather than in parallel.